### PR TITLE
Add network syscalls + more string functions

### PR DESCRIPTION
**Pull request by alexthomas1**
I add network system calls and much needed string util functions.

### CODE REVIEW

File: io_wrap.c [hunk 1]
```diff
@@ -295,4 +297,220 @@ uintptr_t io_syscall_fstatat(int dirfd, char *pathname, struct stat *statbuf,
...
+  edge_syscall->syscall_num = SYS_getcwd;
+
+  size_t totalsize = (sizeof(struct edge_syscall) +
+                      sizeof(sargs_SYS_getcwd));
+
+  dispatch_edgecall_syscall(edge_syscall, totalsize);
+
+  // if(syscall_ret != 0){
+  copy_to_user(buf, &args->buf, size);
+  //   print_strace("[runtime] proxied getcwd\r\n");
```
Comment (dayeol) [inline io_wrap.c:410]: please remove the comments

File: net_wrap.c [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+  args->sockfd = sockfd; 
+  args->addrlen = addrlen; 
+
+  printf("sockfd: %d, addrlen: %d\n", 
+          args->sockfd, args->addrlen);
+
+  copy_from_user(&args->addr, (void *) addr, addrlen);
+
+  printf("sockfd: %d, addrlen: %d\n",
```
Comment (dayeol) [inline net_wrap.c:70]: please remove the prints

File: net_wrap.c [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+  sargs_SYS_bind *args = (sargs_SYS_bind *) edge_syscall->data;
+
+  args->sockfd = sockfd; 
+  args->addrlen = addrlen; 
+
+  printf("sockfd: %d, addrlen: %d\n", 
+          args->sockfd, args->addrlen);
+
+  copy_from_user(&args->addr, (void *) addr, addrlen);
```
Comment (dayeol) [inline net_wrap.c:68]: is there any maxlen for addrlen? if so please add the check
e.g., 
```
if (addrlen > MAXADDRLEN) { return error; }
```

File: io_wrap.c [hunk 4]
```diff
@@ -295,4 +297,220 @@ uintptr_t io_syscall_fstatat(int dirfd, char *pathname, struct stat *statbuf,
...
+           uintptr_t ret = -1;
+  struct edge_syscall* edge_syscall = (struct edge_syscall*)edge_call_data_ptr();
+  sargs_SYS_getpeername* args = (sargs_SYS_getpeername*) edge_syscall->data;
+
+  edge_syscall->syscall_num = SYS_getpeername;
+
+  args->sockfd = sockfd;
+
+  copy_from_user(&args->addrlen, (void *) addrlen, sizeof(socklen_t)); 
+  copy_from_user(&args->addr, (void *) addr, args->addrlen);
```
Comment (dayeol) [inline io_wrap.c:471]: see comment below about max addrlen

File: net_wrap.c [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  edge_syscall->syscall_num = SYS_setsockopt;
+
+  sargs_SYS_setsockopt *args = (sargs_SYS_setsockopt *) edge_syscall->data;
+
+  args->socket = socket; 
+  args->level = level; 
+  args->option_name = option_name; 
+  args->option_len = option_len; 
+
+  copy_from_user(&args->option_value, option_value, option_len);
```
Comment (dayeol) [inline net_wrap.c:41]: please add some minimum check for option_len as well. (max option len)

File: net_wrap.c [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+  uintptr_t ret = -1;
+  struct edge_syscall* edge_syscall = (struct edge_syscall*)edge_call_data_ptr();
+  edge_syscall->syscall_num = SYS_bind;
+
+  sargs_SYS_bind *args = (sargs_SYS_bind *) edge_syscall->data;
+
+  args->sockfd = sockfd; 
+  args->addrlen = addrlen; 
+
+  printf("sockfd: %d, addrlen: %d\n",
```
Comment (dayeol) [inline net_wrap.c:65]: remove print

File: net_wrap.c [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+  uintptr_t ret = -1;
+  struct edge_syscall* edge_syscall = (struct edge_syscall*)edge_call_data_ptr();
+  edge_syscall->syscall_num = SYS_accept;
+
+  sargs_SYS_accept *args = (sargs_SYS_accept *) edge_syscall->data;
+
+  args->sockfd = sockfd; 
+
+  copy_from_user(&args->addrlen, (void *) addrlen, sizeof(socklen_t));
+  copy_from_user(&args->addr, (void *) addr, args->addrlen);
```
Comment (dayeol) [inline net_wrap.c:109]: max addr len

**Review by dayeol — CHANGES_REQUESTED** [review]
please see comments

File: net_wrap.c (hunk 3 above)
Comment (alexthomas1) [reply net_wrap.c:68]: Yes, the max size should be `sizeof(struct sockaddr_storage)`

**Comment by alexthomas1** [comment]
> please see comments

Comments are now addressed. I add max number of bytes copied and removed print statements.

File: io_wrap.c [hunk 8]
```diff
@@ -295,4 +297,193 @@ uintptr_t io_syscall_fstatat(int dirfd, char *pathname, struct stat *statbuf,
...
+
+uintptr_t io_syscall_epoll_create(int size){
+  uintptr_t ret = -1;
+  struct edge_syscall* edge_syscall = (struct edge_syscall*)edge_call_data_ptr();
+  edge_syscall->syscall_num = SYS_epoll_create1;
+
+  sargs_SYS_epoll_create1 *args = (sargs_SYS_epoll_create1 *) edge_syscall->data;
+
+  args->size = size; 
+  printf("epoll-create: %d\n", args->size);
```
Comment (dayeol) [inline io_wrap.c:329]: I missed this one. please remove this as well!

**Review by dayeol — CHANGES_REQUESTED** [review]

**Pull request merged**